### PR TITLE
delete translation static content when parent page is deleted

### PR DESCRIPTION
**Pull request by mjauvin**
Fix for #346

### CODE REVIEW

**Comment by mjauvin** [comment]
Please comment if you think there is a cleaner way to do this or if I am missing something.

File: behaviors/TranslatableCmsObject.php [hunk 1]
```diff
@@ -43,6 +45,20 @@ public function __construct($model)
...
+            $contentPath = sprintf('%s/%s/%s', themes_path(), $model->theme->getDirName(), $model->getObjectTypeDirName());
+
+            foreach (Locale::listEnabled() as $locale => $label) {
+                if ($locale == $this->translatableDefault) {
+                    continue;
+                }
+                $localePath = sprintf('%s-%s/%s', $contentPath, $locale, $model->fileName);
+                if (File::exists($localePath)) {
+                    File::delete($localePath);
+                }
```
Comment (LukeTowers) [inline behaviors/TranslatableCmsObject.php]: Don't use the file facade for this, get the relevant Halcyon object directly and call delete on that; that will ensure support for the upcoming DB layer. Also add a one line comment describing the purpose of this code.

Comment (mjauvin) [reply behaviors/TranslatableCmsObject.php]: This should be the right way

Comment (LukeTowers) [reply behaviors/TranslatableCmsObject.php]: Did you test it?

Comment (LukeTowers) [reply behaviors/TranslatableCmsObject.php]: Why is it required to set the filename if you're calling `getCmsObjectForLocale` already?

Comment (mjauvin) [reply behaviors/TranslatableCmsObject.php]: Yes, tested it and it works fine.

As for setting the filename, I replicated code from another method ( syncTranslatableFileNames ) which does the same thing... but you are right, this is not necessary.

**Comment by mjauvin** [comment]
@mariavilaro @osmanzeki @munxar can someone test this, it's working for me but I prefer having a second opinion.

**Pull request merged**